axum-macros/src/from_request.rs

```diff
@@ -582,26 +602,29 @@ fn extract_fields(
 
         let member = member(field, fields.len() - 1);
         let ty_span = field.ty.span();
-        let into_inner = into_inner(via, ty_span);
+        let into_inner = into_inner(&via, ty_span);
 
         let item = if peel_option(&field.ty).is_some() {
+            let field_ty = into_outer(&via, ty_span, peel_option(&field.ty).unwrap());
             quote_spanned! {ty_span=>
                 #member: {
-                    ::axum::extract::FromRequest::from_request(req, state)
+                    <#field_ty as ::axum::extract::FromRequest<#trait_generics>>::from_request(req, state)
```

Comment:
This should fix the problem you were talking about:

```suggestion
                    <#field_ty as ::axum::extract::FromRequest<#trait_generics, _>>::from_request(req, state)
```